### ISSUE TITLE
Our initial incoming request rate is probably a bad estimate

### ISSUE THREAD

**Issue by rkd-msw** [Metaswitch/sprout#1251]
https://github.com/Metaswitch/sprout/blob/dev/sprout/main.cpp#L1334 sets our initial permitted request rate to 100 requests/sec. This is probably too low: assuming metrics of a re-REGISTER (2 requests) every 5 minutes and a call (~5 requests?) every 30 minutes per subscriber, each subscriber generates about 0.01 requests/second, so we can only handle ~10k subscribers with 100 requests/sec.

Previously, https://github.com/Metaswitch/cpp-common/issues/384 meant that that would creep up from 100 to a higher value, but we've fixed that issue now. To make sure we don't have load problems at start-of-day, we should:

- do some stress testing and figure out how many allowed requests we converge on on a single-core system
- set the initial rate value, as a Sprout command-line argument, to that figure times the number of cores (or maybe 75% of that value, to allow for perf fluctuations/different clock speeds)

**Comment by rkd-msw** [comment]
From testing REGISTER traffic (probably a higher rate of work per request than calls) against an m3.medium Sprout node:

```
18-11-2015 17:39:04.548 UTC Status load_monitor.cpp:251: Maximum incoming request rate/second increased to 594.927612 (based on a smoothed mean latency of 3294 and 0 upstream overload responses)
18-11-2015 17:39:07.550 UTC Status load_monitor.cpp:251: Maximum incoming request rate/second increased to 604.551331 (based on a smoothed mean latency of 3763 and 0 upstream overload responses)
18-11-2015 17:39:10.550 UTC Status load_monitor.cpp:251: Maximum incoming request rate/second increased to 614.146301 (based on a smoothed mean latency of 4050 and 0 upstream overload responses)
18-11-2015 17:39:13.549 UTC Status load_monitor.cpp:251: Maximum incoming request rate/second increased to 623.878601 (based on a smoothed mean latency of 2677 and 0 upstream overload responses)
18-11-2015 17:39:18.547 UTC Status load_monitor.cpp:251: Maximum incoming request rate/second increased to 633.478882 (based on a smoothed mean latency of 3997 and 0 upstream overload responses)
18-11-2015 17:39:21.548 UTC Status load_monitor.cpp:251: Maximum incoming request rate/second increased to 643.256958 (based on a smoothed mean latency of 2219 and 0 upstream overload responses)
18-11-2015 17:39:24.551 UTC Status load_monitor.cpp:251: Maximum incoming request rate/second increased to 652.957581 (based on a smoothed mean latency of 2994 and 0 upstream overload responses)
18-11-2015 17:39:27.549 UTC Status load_monitor.cpp:251: Maximum incoming request rate/second increased to 662.684692 (based on a smoothed mean latency of 2729 and 0 upstream overload responses)
18-11-2015 17:39:31.547 UTC Status load_monitor.cpp:229: Maximum incoming request rate/second decreased to 552.237244 (based on a smoothed mean latency of 924963 and 0 upstream overload responses)
18-11-2015 17:39:34.548 UTC Status load_monitor.cpp:251: Maximum incoming request rate/second increased to 562.198425 (based on a smoothed mean latency of 388 and 0 upstream overload responses)
18-11-2015 17:39:37.547 UTC Status load_monitor.cpp:251: Maximum incoming request rate/second increased to 572.106445 (based on a smoothed mean latency of 920 and 0 upstream overload responses)
18-11-2015 17:39:40.551 UTC Status load_monitor.cpp:251: Maximum incoming request rate/second increased to 581.299622 (based on a smoothed mean latency of 8068 and 0 upstream overload responses)
18-11-2015 17:39:44.550 UTC Status load_monitor.cpp:251: Maximum incoming request rate/second increased to 590.714722 (based on a smoothed mean latency of 5849 and 0 upstream overload responses)
18-11-2015 17:39:49.547 UTC Status load_monitor.cpp:251: Maximum incoming request rate/second increased to 598.792236 (based on a smoothed mean latency of 19225 and 0 upstream overload responses)
18-11-2015 17:39:52.548 UTC Status load_monitor.cpp:251: Maximum incoming request rate/second increased to 607.015259 (based on a smoothed mean latency of 17770 and 0 upstream overload responses)
18-11-2015 17:39:55.548 UTC Status load_monitor.cpp:251: Maximum incoming request rate/second increased to 614.051575 (based on a smoothed mean latency of 29637 and 0 upstream overload responses)
18-11-2015 17:40:01.547 UTC Status load_monitor.cpp:251: Maximum incoming request rate/second increased to 618.898499 (based on a smoothed mean latency of 51531 and 0 upstream overload responses)
18-11-2015 17:40:05.550 UTC Status load_monitor.cpp:229: Maximum incoming request rate/second decreased to 515.748718 (based on a smoothed mean latency of 590561 and 0 upstream overload responses)
```

This suggests that ~600 requests/sec (=300 requests/sec/core) is roughly the rate we converge on, so 250 requests/sec/core would be a sensible default.

**Comment by rkd-msw** [comment]
Fixed in #1266.
